h/schemas/annotation.py

```diff
@@ -409,3 +421,27 @@ class SearchParamsSchema(colander.Schema):
         missing=colander.drop,
         description="Limit the results to annotations made by the specified user.",
     )
+
+    def validator(self, node, cstruct):
+        sort = cstruct['sort']
+        search_after = cstruct.get('search_after', None)
+
+        if search_after:
+            if sort in ["updated", "created"]:
+                self._parsable_date(node, search_after)
+
+            # offset must be set to 0 if search_after is specified.
+            cstruct["offset"] = 0
+
+    def _parsable_date(self, node, value):
```

Comment:
This function name gives the impression it returns a value. Perhaps `_try_parsing_date`?

Reply:
I changed it to _date_is_parsable that returns a True or False and moved the raising to the validate function. I think this separated the concerns a little better.